[더 모니터팀] 백엔드파트 코드리뷰용 PR

JIHYUN2EE opened the pull request:
질문 사항

1. **DB에 방대한 데이터를 저장하게 될 예정인데, 클라이언트로부터 요청이 들어왔을 때 빠르게 데이터를 전달하는 방법에 대해 고민중입니다.**
    1. 과거 스크랩 기사 정보, 보고서 히스토리의 내용들을 담은 DB가 고객사마다 존재해야 하는 상황입니다.
    2. 데이터 관리 효율성을 높일 수 있는 최적의 방식이 있을까요?
2. **Public Search API와 웹 크롤링을 사용했을 때, 실제로 데이터가 응답되기까지 걸리는 딜레이 시간이 얼마나 발생하는지 고민 중입니다.**
    1. API 호출과 크롤링 사이에서 성능 차이가 있는지, 그리고 이를 최적화할 수 있는 방법이 있을지 궁금합니다.

devxb reviewed (commented):
## 질문에 대한 답변

Q: DB에 방대한 데이터를 저장하게 될 예정인데, 클라이언트로부터 요청이 들어왔을 때 빠르게 데이터를 전달하는 방법에 대해 고민중입니다.
과거 스크랩 기사 정보, 보고서 히스토리의 내용들을 담은 DB가 고객사마다 존재해야 하는 상황입니다.
데이터 관리 효율성을 높일 수 있는 최적의 방식이 있을까요?

A: 질문이 2가지로 느껴지는데요. 1. 데이터를 빠르게 응답하는 방법 2. 데이터를 효율적으로 저장하는 방법

우선 무한대의 트래픽과 데이터를 받아내는 서버는 만들 수 없으니, 목표하시는 API latency 가 어느정도인지, 데이터는 얼마나 들어올것이며, 어느정도의 돈을 저장하는데에 사용할 수 있는지 고민해보시면 좋을거 같아요.

데이터를 빠르게 응답하는 방법으로는, 
-실시간성이 어느정도 희생되도 된다면 캐싱을 고려해볼 수 있을거 같습니다. 
-그것이 아니라 준실시간으로 복제지연까지는 타협볼 수 있는 상황이라면 read용 DB를 따로 만들것 같아요.
또, 지금과 같이 테이블이 정규화되어 있을때는 데이터를 조회하기위해서 join이 발생하는데요. read용 DB를 만들고 데이터를 역정규화 시켜놓는다면 join없이 하나의 테이블만 접근해서 원하는 데이터를 뽑아내는 방식으로 성능을 올릴수도 있습니다. 예를들어서, 조회용 mongoDB를 하나 만들고, 모든 조회로직은 mongoDB로 향하게 하고 수정,저장등의 로직은 RDBMS로 향하게 할 수 있겠죠? (이때 수정, 저장이 되었을때 mongoDB로 옮겨주는 로직이 필요합니다.)

데이터를 효율적으로 저장하는 방법으로는, 
상황에 따라 여러가지 방법이 있을거 같은데요. 당장 생각나는 방법은..
스크래핑한지 오래된 데이터는 조회할 확률이 낮아진다고 가정해봅시다. 그렇다면, 일정 시간이 경과한 데이터는 가격대비 용량은 크지만 성능은 좋지않은 저장소로 옮겨서 저장하는 방식을 사용할 수 있을거 같아요. 

## 

Q: Public Search API와 웹 크롤링을 사용했을 때, 실제로 데이터가 응답되기까지 걸리는 딜레이 시간이 얼마나 발생하는지 고민 중입니다. 
API 호출과 크롤링 사이에서 성능 차이가 있는지, 그리고 이를 최적화할 수 있는 방법이 있을지 궁금합니다.

A: (Public Search API가 웹 크롤링 대신 사용하시는, 예를들어 구글이나 네이버에서 제공해주는 api 라고 봐도 되겠죠?) 
Public Search API와 웹 크롤링은 내부적으로 어떤 코드가 구현되어 있을까요? 웹 크롤링에서 더 무거운 동작이 구현되어 있다면 크롤링이 더 느릴거고 반대라면 더 빠를수도 있을겁니다.
속도는 상대적인거 같아서 항상 뭐가 더 빠르다는 정답은 없는거 같아요. ㅎㅎ

(위에서 말씀드렸다 싶이 성능 최적화는 목표치를 설정하고, 도메인 로직을 보면서 가능한 방식의 바람직한 최적화 인지 파악하는게 중요합니다.) 

성능개선을 물어보셔서 시스템 아키텍처? 를 그려봤는데요. 
성능개선을 한다면 아래와 같은 구조로 크롤러 (혹은 API) 파이프라인을 구성해보셔도 재밌을거 같아요.

![스크린샷 2024-11-06 오후 10 02 15](https://github.com/user-attachments/assets/55b34d97-99e3-4ee2-a79d-dbfe1de995e9)

JIHYUN2EE committed:
Merge pull request #76 from The-Monitor-Dev/feature/75-token-check

feature: 토큰 유효성 검사 기능

JIHYUN2EE committed:
Merge pull request #77 from The-Monitor-Dev/develop

feature: 토큰 유효성 검사

L-U-Ready committed:
[✨FEATURE] Implementing Report Management (#78)

* :sparkles: Implement Report Management

report list, delete, detail

Related: #73

* :sparkles: Implement Report Management

separate report title&logo&color and report articles, update title&logo&color

Related: #73

* :sparkles: Implement Report Management

update logo

Related: #73

* :sparkles: Implement Report Management

update logo

Related: #73

L-U-Ready committed:
Merge branch 'develop'

L-U-Ready committed:
:sparkles: Implement Report Management Additional Functions

search Report, delete Report Article

Related: #79

L-U-Ready committed:
:sparkles: Implement Report Management Additional Functions

search Report Order By createdAt DESC

Related: #79

JIHYUN2EE committed:
Merge pull request #80 from The-Monitor-Dev/feature/79-implement-additional-functions-for-report-management

[✨FEATURE] Implement Additional Functions For Report Management

JIHYUN2EE committed:
Merge pull request #82 from The-Monitor-Dev/develop

Merge to main!

L-U-Ready committed:
[✨FEATURE] Implement Report Edit (#84)

* :recycle: Refactor Keyword List Response

Related: #81

* :sparkles: Implement Report Edit Functions

Save Scrap, show scrapArticle by id

Related: #81

* :sparkles: Implement Report Edit Functions

All keyword Search, ReportArticle Summary, ReportArticle Summary Update

Related: #81

* :recycle: Report Response Edit

There is no need to sort Reports

Related: #81

* :recycle:

Related: #81

L-U-Ready committed:
Merge branch 'develop'

JIHYUN2EE committed:
feat: 고객사 입력 수정

JIHYUN2EE committed:
fix: param 어노테이션 수정

JIHYUN2EE committed:
Merge pull request #85 from The-Monitor-Dev/feature/83-keyword-email-settings

feat: 고객사 입력 수정

JIHYUN2EE committed:
Merge pull request #86 from The-Monitor-Dev/develop

Fix: 고객사 카테고리 수정

L-U-Ready committed:
:bug: CategoryType fix

and add public url at securityConfig for test

Related:

L-U-Ready committed:
Merge branch 'develop'

JIHYUN2EE committed:
Feature: 키워드 및 이메일 설정 (#88)

* feat: settings 기능 구현

* feat: settings 기능 구현

* fix: 피드백 수정

* fix: @transcational(readOnly 추가)

L-U-Ready committed:
Merge branch 'develop'

JIHYUN2EE committed:
fix: 오타 및 피드백 수정

JIHYUN2EE committed:
Merge pull request #91 from The-Monitor-Dev/feature/90-fix-wrong-word

fix: 오타 및 피드백 수정

JIHYUN2EE committed:
Merge pull request #92 from The-Monitor-Dev/develop

Merge to Main

L-U-Ready committed:
:rocket: Deploy Files Update (#93)

adapt docker-compose and NGINX

Related: #89

L-U-Ready committed:
Merge branch 'develop'

L-U-Ready committed:
:rocket: Deploy Files Update

Related:

L-U-Ready committed:
:rocket: Deploy Files Update

Related:

L-U-Ready committed:
:rocket: Deploy Files Update

Related:

L-U-Ready committed:
:rocket: Deploy Files Update

Related:

L-U-Ready committed:
:rocket: Deploy Files Update

Related:

L-U-Ready committed:
:rocket: Deploy Files Update

Related:

L-U-Ready committed:
:rocket: Deploy Files Update

Related:

L-U-Ready committed:
:rocket: Deploy Files Update

Related:

L-U-Ready committed:
:rocket: Deploy Files Update

Related:

L-U-Ready committed:
:rocket: Deploy Files Update

Related:

L-U-Ready committed:
:rocket: Deploy Files Update

Related:

L-U-Ready committed:
:rocket: Deploy Files Update

Related:

L-U-Ready committed:
:rocket: Deploy Files Update

Related:

L-U-Ready committed:
:rocket: Deploy Files Update

Related:

L-U-Ready committed:
:rocket: Deploy Files Update

Related:

L-U-Ready committed:
:rocket: Deploy Files Update

Related:

L-U-Ready committed:
:rocket: Deploy Files Update

Related:

L-U-Ready committed:
:rocket: Deploy Files Update

Related:

L-U-Ready committed:
:rocket: Deploy Files Update

Related:

L-U-Ready committed:
:rocket: Deploy Files Update

Related:

L-U-Ready committed:
:rocket: Deploy Files Update

Related:

L-U-Ready committed:
:lock: Domain Cors Config

Related:

L-U-Ready committed:
:rocket: nginx conf update

Related:

L-U-Ready committed:
:rocket: nginx conf update

Related:

L-U-Ready committed:
:rocket: nginx conf update

Related:

L-U-Ready committed:
:rocket: nginx conf update

Related:

L-U-Ready committed:
:rocket: nginx conf update

Related:

L-U-Ready committed:
:rocket: SwaggerConfig update for HTTPS

Related:

JIHYUN2EE committed:
feat: 이미지 업로드 기능 구현

JIHYUN2EE committed:
Merge pull request #95 from The-Monitor-Dev/feature/94-image-upload

feat: 이미지 업로드 기능 구현

JIHYUN2EE committed:
Merge pull request #96 from The-Monitor-Dev/develop

[Feature] Image Upload

L-U-Ready committed:
:rocket: Deploy

Related:

L-U-Ready committed:
:lock: Cookie Config

Related:

L-U-Ready committed:
Merge branch 'develop'

L-U-Ready committed:
:lock: Security Config

Related:

L-U-Ready committed:
Merge branch 'develop'

L-U-Ready committed:
:lock: Security Config

Related:

L-U-Ready committed:
Merge branch 'develop'

L-U-Ready committed:
:bug: Report Create Update

Get Logo Changed

Related:

L-U-Ready committed:
Merge branch 'develop'

L-U-Ready committed:
:bug: Report Create Update

Get Logo Changed

Related:

L-U-Ready committed:
Merge branch 'develop'

L-U-Ready committed:
:bug: Report Create Update

Change Map to dto

Related:

L-U-Ready committed:
Merge branch 'develop'

L-U-Ready committed:
:bug: Report Create Update

Change RequestBody to RequestPart

Related:

L-U-Ready committed:
Merge branch 'develop'

L-U-Ready committed:
:bug: Report Create Update

Related:

L-U-Ready committed:
Merge branch 'develop'

L-U-Ready committed:
:bug: Report Create Update

Related:

L-U-Ready committed:
Merge branch 'develop'

L-U-Ready committed:
:bug: Report Create Update

Related:

L-U-Ready committed:
Merge branch 'develop'

L-U-Ready committed:
:rocket: Nginx.conf update

Related:

L-U-Ready committed:
Merge branch 'develop'

L-U-Ready committed:
:rocket: Nginx.conf update

Related:

L-U-Ready committed:
Merge branch 'develop'

L-U-Ready committed:
:rocket: Nginx.conf update

Related:

L-U-Ready committed:
Merge branch 'develop'

L-U-Ready committed:
:recycle: Report Update Request Type Change (#99)

Request Type RequestParam to RequestBody

Related: #98

L-U-Ready committed:
Merge branch 'develop'

JIHYUN2EE committed:
Feat: 클라이언트 id로 정보 조회

L-U-Ready committed:
:bug: All Keyword Search Article Update (#101)

Adapt OR Operation between keywords

Related: #100

JIHYUN2EE committed:
Merge pull request #103 from The-Monitor-Dev/feature/97-client-info-get

Feat: 클라이언트 id로 정보 조회

JIHYUN2EE committed:
Merge pull request #104 from The-Monitor-Dev/develop

Feat: Get Information using clientId

L-U-Ready committed:
:sparkles: get KeywordList With KeywordId (#105)

Related: #102

L-U-Ready committed:
[🐛 FIX] Resolving Report Generation Errors (#107)

* :bug: Resolving Report Generations Bug

Fix a Bug Which Report Cant Created

Related: #106

* :bug: Resolving Report Generations Bug

Fix a Bug Which Report Cant Created

Related: #106

L-U-Ready committed:
:twisted_rightwards_arrows:

Related:

L-U-Ready committed:
Merge branch 'develop'

JIHYUN2EE committed:
Feat: 클라이언트 정보 수정 및 삭제

JIHYUN2EE committed:
Merge pull request #109 from The-Monitor-Dev/feature/108-client-update

Feat: 클라이언트 정보 수정 및 삭제

JIHYUN2EE committed:
Merge pull request #110 from The-Monitor-Dev/develop

Feat: 고객사 정보 수정 및 삭제

L-U-Ready committed:
:recycle: Change Keyword List Response Format

Related: #111

JIHYUN2EE committed:
Refactor: response body 수정

JIHYUN2EE committed:
Refactor: response body 수정

JIHYUN2EE committed:
Merge pull request #113 from The-Monitor-Dev/refactor/111-change-keyword-list-response-format

[♻️ REFACTOR] Change Keyword List Response Format

JIHYUN2EE committed:
Merge pull request #114 from The-Monitor-Dev/refactor/112-change-response

Refactor: 리스폰스 바디 수정

JIHYUN2EE committed:
Merge pull request #116 from The-Monitor-Dev/develop

Refactor: 리스폰스 수정

L-U-Ready committed:
:bug: Fix Report Search Error (#118)

Fix Error UpdatedAd is Error

Related: #115

L-U-Ready committed:
Merge branch 'develop'

L-U-Ready committed:
Merge remote-tracking branch 'origin/main'

JIHYUN2EE committed:
Feature: 이메일 전송 기능 구현

L-U-Ready committed:
:bug: Fix Report Search Error

Fix Error which search [ can make BadRequest

Related:

L-U-Ready committed:
Merge branch 'develop'

JIHYUN2EE committed:
Merge pull request #119 from The-Monitor-Dev/feature/117-sendmail-with-file

Feature: 이메일 전송 기능 구현

JIHYUN2EE committed:
Merge pull request #121 from The-Monitor-Dev/develop

Feat: 메일 전송하기

JIHYUN2EE committed:
Feature: 고객사 이름 검색 기능 구현

JIHYUN2EE committed:
Merge pull request #123 from The-Monitor-Dev/feature/122-search-client-name

Feature: 고객사 이름 검색 기능 구현

JIHYUN2EE committed:
Merge pull request #124 from The-Monitor-Dev/develop

Feat: 고객사 이름 검색 기능 구현

L-U-Ready committed:
:sparkles: Implement Article Saving Logic

Articles can be saved by Keyword when Client was created

Related: #126

L-U-Ready committed:
:sparkles: Implement new Report Flow

Related: #120

JIHYUN2EE committed:
Merge pull request #127 from The-Monitor-Dev/feature/120-implement-report-category-functions

[✨FEATURE] Implement Report Category Functions

JIHYUN2EE committed:
Merge pull request #129 from The-Monitor-Dev/develop

Implement new Report Flow

JIHYUN2EE committed:
Fix: 서명 엔티티 삭제

JIHYUN2EE committed:
Merge pull request #130 from The-Monitor-Dev/fix/signature-to-client

Fix: 서명 엔티티 삭제

JIHYUN2EE committed:
Merge pull request #131 from The-Monitor-Dev/develop

Fix: 서명 엔티티 삭제

JIHYUN2EE committed:
Fix: 고객사 검색 오류 수정

JIHYUN2EE committed:
Merge pull request #133 from The-Monitor-Dev/fix/132-client-search

Fix: 고객사 검색 오류 수정

JIHYUN2EE committed:
Merge pull request #134 from The-Monitor-Dev/develop

Fix: 고객사 검색 오류 수정

L-U-Ready committed:
[♻️ REFACTOR] Implement Dynamic ClientId Update For Authenticated Accounts

Related: #135

L-U-Ready committed:
Related: #135

JIHYUN2EE committed:
Fix: account에 해당하는 Client를 반환하게 수정

JIHYUN2EE committed:
Merge pull request #136 from The-Monitor-Dev/refactor/135-implement-dynamic-clientid-update-for-authenticated-accounts

Refactor/135 implement dynamic clientid update for authenticated accounts

JIHYUN2EE committed:
Merge pull request #139 from The-Monitor-Dev/fix/138-fix-account-client

Fix: account에 해당하는 Client를 반환하게 수정

JIHYUN2EE committed:
Merge pull request #140 from The-Monitor-Dev/develop

Merge to Main

L-U-Ready committed:
[✨FEATURE] Change Daily Monitoring Logic (#142)

* :sparkles: Article Get Logic update

Related: #137

* :sparkles: Report Update Rollback

Related: #137

L-U-Ready committed:
Merge branch 'develop'

L-U-Ready committed:
:sparkles: Each Keyword List Response Include CategoryType

Related:

L-U-Ready committed:
Merge branch 'develop'

L-U-Ready committed:
:sparkles: ReportArticleUpdateRequest Naming Update

Related:

L-U-Ready committed:
Merge branch 'develop'

L-U-Ready committed:
:sparkles: ReportArticleUpdate Logic Update

Related:

L-U-Ready committed:
Merge branch 'develop'

L-U-Ready committed:
:sparkles:

Related:

L-U-Ready committed:
Merge branch 'develop'

JIHYUN2EE committed:
Fix: 프론트 요구사항 수정

JIHYUN2EE committed:
Feat: 엑셀 파일

JIHYUN2EE committed:
Merge pull request #143 from The-Monitor-Dev/fix-front

Fix: 프론트 요구사항 수정

JIHYUN2EE committed:
Merge pull request #144 from The-Monitor-Dev/develop

FIX: 프론트 요구사항 수정

L-U-Ready committed:
:sparkles:

Related:

L-U-Ready committed:
Merge branch 'develop'

L-U-Ready committed:
:sparkles:

Related:

L-U-Ready committed:
Merge branch 'develop'

L-U-Ready committed:
:sparkles:

Related:

L-U-Ready committed:
Merge branch 'develop'

JIHYUN2EE committed:
Feat: 스크랩 기능 수정

JIHYUN2EE committed:
Merge branch 'develop' into feature/141-change-scrap

L-U-Ready committed:
:sparkles:

Related:

L-U-Ready committed:
Merge branch 'develop'

JIHYUN2EE committed:
Merge pull request #145 from The-Monitor-Dev/feature/141-change-scrap

Feat: 스크랩 기능 수정

JIHYUN2EE committed:
Merge pull request #146 from The-Monitor-Dev/develop

Feat: 스크랩 기능

JIHYUN2EE committed:
Merge branch 'develop' into feature/125-create-report-excel

JIHYUN2EE committed:
Fix: 보고서 ReportListResponse 수정

JIHYUN2EE committed:
특정 커밋에서 수정된 파일만 푸쉬

JIHYUN2EE committed:
Merge pull request #147 from The-Monitor-Dev/feature/125-create-report-excel

Fix: response 수정

JIHYUN2EE committed:
Merge pull request #148 from The-Monitor-Dev/develop

Fix: response 수정

JIHYUN2EE committed:
빠진 파일 추가

JIHYUN2EE committed:
Merge pull request #149 from The-Monitor-Dev/develop

빠진 파일 추가

JIHYUN2EE committed:
Response 변경

JIHYUN2EE committed:
Merge pull request #150 from The-Monitor-Dev/develop

Response 변경

L-U-Ready committed:
:sparkles:

Related:

L-U-Ready committed:
Merge branch 'develop'

L-U-Ready committed:
:sparkles:

Related:

L-U-Ready committed:
Merge branch 'develop'

L-U-Ready committed:
:sparkles: Report Category CRUD

Related:

L-U-Ready committed:
Merge branch 'develop'

JIHYUN2EE committed:
fix: 키워드 업데이트 수정

JIHYUN2EE committed:
Merge pull request #152 from The-Monitor-Dev/develop

fix: 키워드 업데이트 수정

L-U-Ready committed:
:sparkles: Article Saving Improved

Related:

L-U-Ready committed:
Merge branch 'develop'

L-U-Ready committed:
:sparkles: Article Saving Improved

Related:

L-U-Ready committed:
:sparkles: Article Saving Improved

Related:

L-U-Ready committed:
Merge branch 'develop'

L-U-Ready committed:
:sparkles:

Related:

L-U-Ready committed:
Merge branch 'develop'

L-U-Ready committed:
:sparkles:

Related:

L-U-Ready committed:
Merge branch 'develop'

L-U-Ready committed:
:sparkles:

Related:

L-U-Ready committed:
Merge branch 'develop'

L-U-Ready committed:
:sparkles:

Related:

L-U-Ready committed:
Merge branch 'develop'

L-U-Ready committed:
:sparkles:

Related:

L-U-Ready committed:
Merge branch 'develop'

L-U-Ready committed:
:sparkles:

Related:

L-U-Ready committed:
Merge branch 'develop'

JIHYUN2EE committed:
Fix: 스크랩 로직 수정

JIHYUN2EE committed:
Merge pull request #153 from The-Monitor-Dev/feature/151-email-with-excel

Fix: 스크랩 로직 수정

JIHYUN2EE committed:
Merge pull request #154 from The-Monitor-Dev/develop

Fix: 스크랩 로직 수정

JIHYUN2EE committed:
Feat: 엑셀 추출 및 이메일 전송 (#155)

JIHYUN2EE committed:
Merge pull request #156 from The-Monitor-Dev/develop

Feat: 엑셀 추출 및 이메일 전송 (#155)

JIHYUN2EE committed:
에러 수정

JIHYUN2EE committed:
merge branch 'develop' of https://github.com/The-Monitor-Dev/The-Monitor_BE into develop
the commit.

JIHYUN2EE committed:
에러 수정

JIHYUN2EE committed:
Merge pull request #157 from The-Monitor-Dev/fix-error

Fix error

JIHYUN2EE committed:
Merge pull request #158 from The-Monitor-Dev/develop

Fix error

L-U-Ready committed:
:sparkles: save

Related:

L-U-Ready committed:
Merge branch 'develop'

L-U-Ready committed:
:sparkles: Report Options Logic

Related:

L-U-Ready committed:
Merge branch 'develop'

L-U-Ready committed:
:sparkles: Publisher and PublishDate Null Check

Related:

L-U-Ready committed:
Merge branch 'develop'

L-U-Ready committed:
:sparkles: Publisher and PublishDate Null Check

Related:

L-U-Ready committed:
Merge branch 'develop'

JIHYUN2EE committed:
Feat: 엑셀 추출 및 이메일 전송

JIHYUN2EE committed:
Merge pull request #159 from The-Monitor-Dev/feat/excel-email

Feat: 엑셀 추출 및 이메일 전송

JIHYUN2EE committed:
Merge pull request #160 from The-Monitor-Dev/develop

Feat: 이메일 내보내기

L-U-Ready committed:
:sparkles:

Related:

L-U-Ready committed:
Merge branch 'develop'

L-U-Ready committed:
:sparkles:

Related:

L-U-Ready committed:
Merge branch 'develop'

L-U-Ready committed:
:sparkles:

Related:

L-U-Ready committed:
Merge branch 'develop'

L-U-Ready committed:
:sparkles:

Related:

L-U-Ready committed:
Merge branch 'develop'

L-U-Ready committed:
:sparkles:

Related:

JIHYUN2EE committed:
Feat: 키워드 업데이트 시, 기사 새로 저장

JIHYUN2EE committed:
Merge pull request #161 from The-Monitor-Dev/feat/keyword-update

Feat: 키워드 업데이트 시, 기사 새로 저장

L-U-Ready committed:
:sparkles: BaseTimeEntity Update Format

Related:

L-U-Ready committed:
:sparkles: GoogleArticle publishDate Update Format

Related:

L-U-Ready committed:
:sparkles: Scrap Response and ReportArticle Keyword Update

Related:

L-U-Ready committed:
:sparkles: Scrap Response and ReportArticle Keyword Update

Related:

L-U-Ready committed:
:sparkles:

Related:

JIHYUN2EE committed:
Fix: 이메일 전송

JIHYUN2EE committed:
Merge pull request #163 from The-Monitor-Dev/fix/excel-style

Fix: 엑셀 스타일 변경

L-U-Ready committed:
:sparkles:

Related:

L-U-Ready committed:
Merge remote-tracking branch 'origin/develop' into develop

L-U-Ready committed:
:sparkles:

Related:

L-U-Ready committed:
:sparkles:

Related:

L-U-Ready committed:
:sparkles:

Related:

L-U-Ready committed:
:sparkles: IsAdded implemented

Related:

L-U-Ready committed:
:sparkles: IsRead Implemented

Related:

L-U-Ready committed:
:sparkles: IsRead Implemented

Related:

L-U-Ready committed:
:bug: Fix Scrap

Related:

L-U-Ready committed:
:bug: Fix Scrap

Related:

JIHYUN2EE committed:
Fix: 엑셀 스타일 수정

L-U-Ready committed:
:bug: Fix Scrap

Related:

JIHYUN2EE committed:
Merge branch 'develop' into fix/excel-style-change

JIHYUN2EE committed:
Fix: 엑셀 스타일 수정

JIHYUN2EE committed:
Fix: 엑셀 스타일 수정

JIHYUN2EE committed:
Merge pull request #165 from The-Monitor-Dev/fix/excel-style-change

Fix: excel style change

JIHYUN2EE committed:
Fix: 엑셀 스타일 수정

L-U-Ready committed:
:sparkles:

Related:

L-U-Ready committed:
Merge remote-tracking branch 'origin/develop' into develop

L-U-Ready committed:
:wastebasket:

Related:

L-U-Ready committed:
:ambulance:

Related:

L-U-Ready committed:
:ambulance:

Related:

L-U-Ready committed:
:wastebasket:

Related:

JIHYUN2EE committed:
Fix: 엑셀 날짜 수정
